weatherman.py

```diff
@@ -0,0 +1,213 @@
+import os
+import sys
+import calendar
+
+highest_temp= ['date',-50]                     # highest_temp[0] would contain date and temp[1] would contain max_temp on that day
+lowest_temp= ['date',50]
+highest_humidity=['date',-50]
+
+
+# def user_input():
+
+#     year=input()
+#     return year
+
+
+
+
+def file_name_matching(year,month_name):                         #send null string in case of ingle argument
+
+     for file_name in os.listdir('.'):
+
+        if month_name=="":
+            if fnmatch.fnmatch( file_name, '*'+year+'*'):        # check wether file name matches with pattern given as second argument. '*' means could be anything
+                file_parsing_year(file_name)
+
+        elif fnmatch.fnmatch(file_name, '*'+year+'_'+month_name[:3]+'*'):  #[:3] because file name only consist 3 alphabets of month_name
+            return file_name
+
+
+
+
+def file_parsing_year(file_name):
+
+    global highest_temp, lowest_temp,highest_humidity             #make variables global
+
+    file_open=open(file_name,"r")
+    lines = file_open.readlines()[1:]                             # read file after 1 lines. It will return a list containing each line in filecending with \n
+ 
+    for line in lines:                                             # Iterate through each line
+        values = line.strip().split(',')                           # Strip any leading/trailing whitespace (like newlines) and split by comma
+       
+        if(values[3]!="" and int(values[3])<lowest_temp[1] ):       # values[3]!="" because "empty" cann't be converted to int
+                                                            
+            lowest_temp[1]=int(values[3])
+            lowest_temp[0] =values[0]
+        
+        if(values[1]!=""  and int(values[1])>highest_temp[1]  ):   # values[1]!="" because "empty" cann't be converted to int
+            
+           highest_temp[1]=int(values[1])
+           highest_temp[0] =values[0]
+    
+        if(values[7]!="" and int(values[7])>highest_humidity[1]):
+
+            highest_humidity[1]=int(values[7])
+            highest_humidity[0] =values[0]
+        
+    file_open.close()
+    
+
+
+def file_parsing_month(file_name):
+
+    average_values=[0,0,0]                                           # list avg_max_temp. avg_min temp, avg_mean_humidity
+    file_open=open(file_name,"r")
+    lines = file_open.readlines()[1:] 
+    total_lines=1
+    for line in lines:                                               # Iterate through each line
+        total_lines+=1                                               #to calculate total lines in a file
+        values = line.strip().split(',')                             # Strip any leading/trailing whitespace (like newlines) and split by comma
+        if(values[1]!=""):
+            average_values[0]+=int(values[1])
+        if(values[3]!=""):
+            average_values[1]+=int(values[3])
+        if(values[8]!=""):
+            average_values[2]+=int(values[8])
```

Comment:
What if order of data is changed? Let's say 8th index has highest temperature instead of 1? Think of a better approach.